### PR TITLE
deploy: use subPath in hooks for volumes when defined

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/15445

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

@kargakis or @tnozicka quick review pls.

**Comment by mfojtik** [comment]
i guess we should have a fuzzer test for volumes here to make sure we copy all fields (if there are any new fields added so we won't miss things like this in the future)

File: pkg/deploy/strategy/support/lifecycle.go [hunk 1]
```diff
@@ -373,6 +373,7 @@ func makeHookPod(hook *deployapi.LifecycleHook, rc *kapi.ReplicationController,
 				Name:      mount.Name,
 				ReadOnly:  mount.ReadOnly,
 				MountPath: mount.MountPath,
+				SubPath:   mount.SubPath,
```
Comment (tnozicka) [inline pkg/deploy/strategy/support/lifecycle.go]: Can't we just copy the whole object? there should be a deep copy method generated somewhere, right?

Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: we can, but for now I think it is safer to list the options we want to copy :)

Comment (tnozicka) [inline pkg/deploy/strategy/support/lifecycle.go]: it would mitigate the need for your fuzzer test :)

**Comment by tnozicka** [comment]
lgtm

**Comment by mfojtik** [comment]
@openshift-ci-robot watch this:

[merge]

File: pkg/deploy/strategy/support/lifecycle.go (hunk 1 above)
Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: yeah, can't decide what would be better :-)

**Comment by stevekuznetsov** [comment]
@kargakis seeing `Timed out waiting for PR 15447 to finish testing` in the sq logs for this one

**Comment by kargakis** [comment]
flaked on https://github.com/openshift/origin/issues/8502

**Comment by kargakis** [comment]
```
package openshift-ansible-3.6.154-1.git.0.2c7a127.el7 is not installed
```

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/15506

**Comment by mfojtik** [comment]
@kargakis 7 of 11 checks passed and still merged? sounds scary :-)

**Comment by kargakis** [comment]
@mfojtik this was merged as part of a batch job where all checks passed so you have to be thankful that you are not still fighting to get this merged. :)

**Comment by kargakis** [comment]
Btw the only failed context is from the old job set by Rosie. The unset contexts are actually successful runs but the PR was merged before they were updated by prow (since the batch job was faster than the single job).

https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_integration/5028/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install_update/3020/

**Pull request merged**